src/StableHashTraits.jl

```diff
@@ -195,7 +207,7 @@ define the one-argument version of `hash_method` and/or two argument version of
 
 ```
 """
-hash_method(::Any) = UseWrite()
+hash_method(x::Any) = Tables.istable(x) ? UseTable() : UseWrite()
```

Comment:
My suggestion above would be something like:
```suggestion
function hash_method(x::Any)
    applicable(write, x) && return UseWrite()
    if Tables.istable(x)
        return IsTable()
    else
        throw(ArgumentError("`Base.write` does not have a method applicable to the input of type $(typeof(x)). Additionally `Tables.istable` returns false for this object. Therefore one must define `hash_method(x::$(typeof(x))` in order for StableHashTraits to know how to hash this object. If this type belongs to Base or a standard library, consider filing an issue or PR to add the trait to StableHashTraits.jl itself."))
    end
end
```

Reply:
Nice, with this change, I guess this PR wouldn't be breaking would it 🤔

Reply:
Unfortunately:

```
julia> struct MyType end
julia>
julia> applicable(Base.write, IOBuffer(), MyType())
true
julia> applicable(StableHashTraits.write, IOBuffer(), MyType())
true
```

I don't think this would work...